crosschain-resolver/contracts/L1Resolver.sol

```diff
@@ -0,0 +1,131 @@
+        EVMFetcher.newFetchRequest(verifier, target)
+            .getStatic(RECORD_VERSIONS_SLOT)
+              .element(node)
+            .getDynamic(VERSINABLE_ADDRESSES_SLOT)
+              .ref(RECORD_VERSIONS_REF)
+              .element(node)
+              .element(COIN_TYPE_ETH)
+            .fetch(this.addrCallback.selector, ''); // recordVersions
+    }
+
+    function addrCallback(
+        bytes[] memory values,
+        bytes memory
+    ) public pure returns (address) {
+        return bytesToAddress(values[1]);
+    }
+
+    /**
+     * Returns the address associated with an ENS node.
+     * @param node The ENS node to query.
+     * @param coinType The cointype to query
+     * @return The associated address.
+     */
+    function addr(
+        bytes32 node,
+        uint256 coinType
+    ) public view returns (bytes memory) {
+        EVMFetcher.newFetchRequest(verifier, target)
+            .getStatic(RECORD_VERSIONS_SLOT)
+              .element(node)
+            .getDynamic(VERSINABLE_ADDRESSES_SLOT)
+              .ref(RECORD_VERSIONS_REF)
+              .element(node)
+              .element(coinType)
+            .fetch(this.addrCoinTypeCallback.selector, '');
+    }
+
+    function addrCoinTypeCallback(
+        bytes[] memory values,
+        bytes memory
+    ) public pure returns (bytes memory) {
+        return values[1];
+    }
+
+    /**
+     * Returns the text data associated with an ENS node and key.
+     * @param node The ENS node to query.
+     * @param key The text data key to query.
+     * @return The associated text data.
+     */
+    function text(
+        bytes32 node,
+        string calldata key
+    ) public view returns (string memory) {
+        EVMFetcher.newFetchRequest(verifier, target)
+            .getStatic(RECORD_VERSIONS_SLOT)
+              .element(node)
+            .getDynamic(VERSINABLE_TEXTS_SLOT)
+              .ref(RECORD_VERSIONS_REF)
+              .element(node)
+              .element(key)
+            .fetch(this.textCallback.selector, '');
+    }
+
+    function textCallback(
+        bytes[] memory values,
+        bytes memory
+    ) public pure returns (string memory) {
+        return string(values[1]);
+    }
+
+    /**
+     * Returns the contenthash associated with an ENS node.
+     * @param node The ENS node to query.
+     * @return The associated contenthash.
+     */
+    function contenthash(bytes32 node) public view returns (bytes memory) {
+        EVMFetcher.newFetchRequest(verifier, target)
+            .getStatic(RECORD_VERSIONS_SLOT)
+              .element(node)
+            .getDynamic(VERSINABLE_HASHES_SLOT)
+              .ref(RECORD_VERSIONS_REF)
+              .element(node)
+            .fetch(this.contenthashCallback.selector, '');
+    }
+
+    function contenthashCallback(
+        bytes[] memory values,
+        bytes memory
+    ) public pure returns (bytes memory) {
+        return values[1];
+    }
+
+    function bytesToAddress(
```

Comment:
I believe you can just cast the `bytes` to a `bytes20` rather than using assembly.